students/Josh_HOff/lesson08/circle_class.py

```diff
@@ -0,0 +1,134 @@
+import sys
+import copy
+import math
+import pathlib
+from functools import total_ordering
+
+circles = []
+@total_ordering
+class Circle(object):
+
+    def __init__(self, value):
+        global circles
+        global templist
+        self._radius = value
+        self._diameter = value * 2
+        self._area = (math.pi) * (value ** 2)
+    
+    @staticmethod
+    def from_diameter(value):
+        return Circle(value/2)
+        
+    @property
+    def radius(self):
+        return self._radius
+    @radius.setter
+    def radius(self, value):
+        self._radius = value
+        self._diameter = value * 2
+        self._area = (math.pi) * (value ** 2)
+    @property
+    def diameter(self):
+        return self._diameter
+    @diameter.setter
+    def diameter(self, value):
+        self._diameter = value
+        self._radius = value / 2
+        self._area = (.25 * math.pi) * (value ** 2)
+    @property
+    def area(self):
+        return self._area
+    @area.setter
+    def area(self, value):
+        raise AttributeError
```

Comment:
if you don't define a `setter` then it will error anyway

Reply:
@natasha-aleksandrova So I could just delete lines 41 through 43 and it will work essentially the same?

Reply:
correct; if attribute doesn't exist it will throw the same error:
```
>>> class Foo: pass
...
>>> Foo().foo
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Foo' object has no attribute 'foo'
>>>
```